research-software-directory/RSD-as-a-service
# 104 add software inital step

# Add software - inital step - provide name and short description

Closes #104 

Changes proposed in this pull request:

*   The user is able to add new software after loggin in with SURFconext test credentials.   

How to test:

* `docker-compose up` will build new version of frontend image and start RSD on localhost. In this case data-migration part is not relevant (testing does not require software in the database).
* before you sign in, you will not be able to add software, the plus button (top right corner) will not be visible and the software/add route should produce 401 message.
* After signing in you will see + button. Clicking will show "New software" option. To sign in, select Test IdP and any of dummy users (for example, "professor 1")
![image](https://user-images.githubusercontent.com/9204081/151351605-35cea2de-5f08-431d-bf17-c42a2422f963.png)
* After clicking on "New software", the modal will appear. Provide minimal information: name and short description
![image](https://user-images.githubusercontent.com/9204081/151351768-13ba26b8-4f59-4f17-a652-8beb76f284c7.png)
* You can cancel/close the process by using ESC button, Cancel button or clicking outside of the modal (default behavior of the modal component)
* During typing you will get information about "validity" of the entry. Both fields are required and have min/max defined as:
   * name: min=3, max=100
   * short description: min=10, max=300 
![image](https://user-images.githubusercontent.com/9204081/151353410-995cdce2-a22a-4685-9995-baab446e3385.png)
* Save button is disabled at the start and will be enabled when all (required) input is according to the requirements.
* Click on the save button will create software and forward you to edit page, where the rest of the information should be provided (will be developed later).
    *  At the first section, we now only show saved data in JSON format for validation purposes. You can navigate between edit page sections using the left panel. The section will only have title so you can confirm that the section components are loading and navigation works. 
    * The plus button (top right) you used to add new software is disabled at this stage, because you are in the process of creating/editing an software item. 
    * Save button (top right) is not implemented (will console log "TODO!..."). 
    * Back button sends you back to a page you were at the moment of pressing new software menu option.
![image](https://user-images.githubusercontent.com/9204081/151354765-10109332-7359-41d5-a185-cc8b7e846fdf.png)
* Providing the software with the identical name second time will cause error and the following message will be shown
![image](https://user-images.githubusercontent.com/9204081/151354107-ec473c56-4e57-44c8-85f7-77320bd79bcb.png)

PR Checklist:
